packages/connect/src/index.ts

```diff
@@ -1,3 +1,101 @@
+ * In the library's internals, {createPolkadotJsScClient} is implemented on top of
+ * {getConnectorClient}.
+ *
+ * If you use {createPolkadotJsScClient}, be aware that the PolkadotJS library and its API are
+ * fundamentally built around full nodes functionnalities, while substrate-connect is a light
+ * client. Some functionnalities of PolkadotJS will not work properly, and/or not as well as they
+ * could. This is fundamentally not fixable, and, while {createPolkadotJsScClient} is acceptable
+ * for demos and prototypes, it is not possible to achieve the best user experience by using
+ * PolkadotJS. Proper light-client-oriented high-level libraries built on top of
+ * {getConnectorClient} are currently in development.
+ *
+ * # Adding parachains
+ *
+ * Connecting to a parachain is done the same way as connecting to a standalone chain: obtaining
+ * a client (with {getConnectorClient} or {createPolkadotJsScClient}) then calling `addChain`.
+ *
+ * However, if you call `addChain` with a parachain chain specification, you **must** have
+ * connected to its corresponding relay chain beforehand (using `addChain` or `addWellKnownChain`).
+ * Failing to do so will lead to an error at the initialization of the parachain.
+ *
+ * Furthermore, the parachain must be added to the same client object as the one the relay chain
+ * was added to.
+ *
+ * In other words, this will work:
+ *
+ * ```js
+ * const client = getConnectorClient();
+ * await client.addChain(relayChain);
+ * await client.addChain(parachain);
+ * ```
+ *
+ * While this will **not** work, and an exception will be thrown when adding the parachain:
+ *
+ * ```js
+ * await getConnectorClient().addChain(relayChain);
+ * await getConnectorClient().addChain(parachain);
+ * ```
+ *
+ * # Resources sharing
+ *
+ * While calling {getConnectorClient} or {createPolkadotJsScClient} multiple times leads to a
+ * different observable behaviour when it comes to parachains (see previous section), internally
+ * resources are shared between all the clients.
+ *
+ * In order words, it is not a problem to do this:
+ *
+ * ```js
+ * const relayChain = ...;
+ * const chain1 = await getConnectorClient().addChain(relayChain);
+ * const chain2 = await getConnectorClient().addChain(relayChain);
+ * ```
+ *
+ * From an API perspective, `chain1` and `chain2` should be treated as two completely separate
+ * connections to the same chain. Internally, however, only one "actual" connection to that chain
+ * will exist.
+ *
+ * This means that there is no problem in calling {getConnectorClient} or
+ * {createPolkadotJsScClient} from within a library for example.
+ *
+ * # Well-known chains
+ *
+ * This package contains a list of so-called {WellKnownChains}. This is a list of popular chains
+ * that users are likely to connect to. Instead of calling `addChain` with a chain specification,
+ * one can call `addWellKnownChain`, passing only the name of a well-known chain as parameter.
+ *
```

Comment:
```suggestion
 * At the moment these are polkadot, kusama, rococo and westend.
 * See <https://github.com/paritytech/substrate-connect/blob/main/packages/connect/src/WellKnownChains.ts>.
```

Reply:
Normally the JS docs generator should link to `WellKnownChains`, there's no reason to put a link.